### ISSUE TITLE
`bugwarrior-pull --dry-run` fails with not-yet-authed gmail integration

### ISSUE THREAD

**Issue by pjf** [ralphbean/bugwarrior#488]
I'm just getting started with gmail integration. If I run `bugwarrior-pull`, it bounces me over to a Google Accounts sign-in after reading my credentials file. However if I run `bugwarrior-pull --dry-run` I get:

```
INFO:bugwarrior.services:Working on [gmail]
INFO:bugwarrior.services.gmail:Starting authentication for gmail
INFO:bugwarrior.services.gmail:No valid login. Starting OAUTH flow.
usage: bugwarrior-pull [--auth_host_name AUTH_HOST_NAME]
                       [--noauth_local_webserver]
                       [--auth_host_port [AUTH_HOST_PORT [AUTH_HOST_PORT ...]]]
                       [--logging_level {DEBUG,INFO,WARNING,ERROR,CRITICAL}]
bugwarrior-pull: error: unrecognized arguments: --dry-run
CRITICAL:bugwarrior.services:2
INFO:bugwarrior.services:Done with [gmail] in 0.016221s
INFO:bugwarrior.services:Terminating workers
CRITICAL:bugwarrior.command:Aborted (critical error in target 'gmail') 
```

<s>I don't know if dry-runs will work after I've finished the oauth dance (testing that next), but in an unauthenticated state the error isn't what one would expect. :)</s>

`--dry-run` works fine if authentication has been completed. :)

Relates to #477, so cc'ing in the fabulous @cthulahoops. Thank you *so much* for implementing this; I feel I may be finally clearing my inbox sometime soon!

**Comment by pjf** [comment]
I'm guessing desired behaviours with `--dry-run` could be either:

- Tell the user it's not possible to use `--dry-run` if we're not authed, or
- Do the auth (even saving it), but not updating/inserting any tasks.

I'd feel the second option would make the most sense, because it allows the user to play with different options before committing in any way.

More exciting, *it's working*. I now have my pinned gmail tasks in TaskWarrior! :D :D :D

**Comment by cthulahoops** [comment]
Yeah, I think the second would be better too. I don't think it should be too difficult. I hadn't tried dry-run. I'll have a look and see what's involved.

Very glad this is useful to you. :)

**Comment by pjf** [comment]
> Very glad this is useful to you. :)

@cthulahoops : It's *amazing*. Seriously, thank you *so* much.

**Comment by ryneeverett** [comment]
Also a vote for the second option. The point of the `--dry-run` flag is to do everything other than modify the taskwarrior database.

**Comment by ryneeverett** [comment]
Resolved by #491.
